delivery/src/main/java/com/storyous/delivery/common/DeliveryModel.kt

```diff
@@ -57,22 +59,35 @@ open class DeliveryModel : CoroutineScope by CoroutineProviderScope() {
         }
     }
 
-    fun loadOrders(): Job? {
+    fun loadOrders(): Job {
         if (loadingOrdersJob?.isActive != true) {
             loadingOrdersJob = launch(provider.Main) {
-                DeliveryConfiguration.placeInfo?.let {
-                    Timber.d("Delivery order download started.")
+                val (placeId, merchantId, autoConfirmEnabled) = DeliveryConfiguration.placeInfo.also {
+                    if (it == null) Timber.w("Delivery is not configured to load new orders")
+                } ?: return@launch
+
+                Timber.d("Delivery order download started.")
+
+                val alreadyConfirmedOrders = DeliveryConfiguration.deliveryRepository
+                    .takeIf { autoConfirmEnabled }
+                    ?.getConfirmedOrdersFromDb()
+                    ?.map { it.orderId }
 
-                    DeliveryConfiguration.deliveryRepository
-                        ?.loadDeliveryOrders(it.merchantId, it.placeId)
+                val loadDeliveryOrders = DeliveryConfiguration.deliveryRepository
```

Comment:
Nejdřív jsem přemejšlel, proč to s tím dalším blokem není zřetězený, ale to už bylo asi moc velký, co?
Možná bych to tady jen pojmenoval třeba ordersForAutoConfirm nebo tak něco.

Reply:
napsal jsem to v jednom bloku nakonec